src/server/scripts/KulTiras/ShrineOfTheStorm/boss_aqusirr.cpp

```diff
@@ -0,0 +1,601 @@
+        instance->SetBossState(DATA_AQUSIRR, FAIL);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_GRASP_FROM_THE_DEPTHS_DAMAGE);
+
+        summons.DespawnAll();
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        instance->SetBossState(DATA_AQUSIRR, DONE);
+
+        Creature* shrineDummy = GetClosestCreatureWithOptions(me, 50.0f, { .CreatureId = NPC_SHRINE_OF_THE_STORM_DUMMY, .IgnorePhases = true });
+        if (!shrineDummy)
+            return;
+
+        shrineDummy->CastSpell(shrineDummy, instance->instance->GetTeamInInstance() == HORDE ? SPELL_CONVERSATION_HORDE : SPELL_CONVERSATION_ALLIANCE, true);
+
+    }
+
+    void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+    {
+        if (me->HealthBelowPctDamaged(50, damage) && !_triggered)
+        {
+            _triggered = true;
+            me->SetUnkillable(true);
+            DoCastSelf(SPELL_ERUPTING_WATERS_SPLIT);
+        }
+    }
+
+    void SpellHit(WorldObject* /*caster*/, SpellInfo const* spellInfo) override
+    {
+        if (spellInfo->Id == SPELL_ERUPTING_WATERS_AQUALING)
+        {
+            if (_aqualingDefeatedCount <= 1)
+            {
+                DoCastSelf(SPELL_ERUPTING_WATERS_DAMAGE);
+                _aqualingDefeatedCount++;
+            }
+            else if (_aqualingDefeatedCount == 2)
+            {
+                DoCastSelf(SPELL_ERUPTING_WATERS_DAMAGE);
+                me->RemoveAurasDueToSpell(SPELL_ERUPTING_WATERS_SPLIT);
+                me->SetUnkillable(false);
+            }
+        }
+    }
+
+private:
+    bool _triggered;
+    uint8 _aqualingDefeatedCount;
+};
+
+// 134828 - Aqualing
+struct boss_aqusirr_aqualing : public AqusirrSharedAI
+{
+    boss_aqusirr_aqualing(Creature* creature) : AqusirrSharedAI(creature, DATA_AQUALING) { }
+
+    void JustSummoned(Creature* summon) override
+    {
+        summons.Summon(summon);
+        DoZoneInCombat();
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        DoCast(SPELL_ERUPTING_WATERS_AQUALING);
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        AqusirrSharedAI::JustEngagedWith(who);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me, 2);
+        ScheduleEvents();
+    }
+};
+
+// 139737 - Lord Stormsong
+struct npc_aqusirr_intro_lord_stormsong : public CreatureAI
+{
+    npc_aqusirr_intro_lord_stormsong(Creature* creature) : CreatureAI(creature) { }
+
+    void DoAction(int32 action) override
+    {
+        if (action == ACTION_START_INTRO)
+        {
+            me->RemoveAurasDueToSpell(SPELL_WATER_RITUAL);
+
+            _scheduler.Schedule(1s, [this](TaskContext context)
```

Comment:
dont try to build spaceships, use a variable `Milliseconds delay =1s;` and then use it as value for scheduling

Reply:
i like spaceships :(